iOS/Source/MainController.swift

```diff
@@ -1,121 +1,236 @@
+
+        return button
     }()
 
-    lazy var locationTracker: LocationTracker = {
-        let tracker = LocationTracker()
-        tracker.delegate = self
+    private lazy var sunView: SunView = { SunView() }()
 
-        return tracker
+    private lazy var messageLabel: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 0
+        label.font = Theme.light(size: 32)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = .white
+        label.minimumScaleFactor = 0.3
+        label.adjustsFontSizeToFitWidth = true
+
+        return label
     }()
 
-    fileprivate var rootView: MainView {
-        return self.view as! MainView
-    }
+    private lazy var locationLabel: UILabel = {
+        let label = UILabel()
+        label.font = Theme.light(size: 16)
+        label.translatesAutoresizingMaskIntoConstraints = false
 
-    override func loadView() {
-        let view = MainView(frame: UIScreen.main.bounds)
-        view.delegate = self
-        self.view = view
-    }
+        return label
+    }()
 
-    override var preferredStatusBarStyle: UIStatusBarStyle {
-        return .default
+    private lazy var shareButton: UIButton = {
+        let button = UIButton(type: .custom)
+        button.setTitle("Share", for: .normal)
+        button.titleLabel?.font = Theme.light(size: 16)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(didSelectShareButton(button:)), for: .touchUpInside)
+
+        return button
+    }()
+
+    private var informationButtonTopAnchor: NSLayoutConstraint?
+    private var locationLabelBottomAnchor: NSLayoutConstraint?
+    private var locationLabelLeftAnchor: NSLayoutConstraint?
+    private var locationLabelRightAnchor: NSLayoutConstraint?
+
+    init(withDaylightModelController dayLightModelController: DaylightModelController) {
+        self.dayLightModelController = dayLightModelController
+        super.init(nibName: nil, bundle: nil)
+
+        self.dayLightModelController.delegate = self
     }
 
-    override var prefersStatusBarHidden: Bool {
-        return true
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        self.locationTracker.locateIfPossible()
-        self.updateInterface()
+        self.view.backgroundColor = .blue
+        self.addSubviewsAndConstraints()
 
-        Timer.scheduledTimer(timeInterval: 30, target: self, selector: #selector(self.updateInterface), userInfo: nil, repeats: true)
-
-        Notifier.cancelAllNotifications()
-        if Settings.areNotificationsEnabled {
-            if let location = Location.current {
-                Notifier.scheduleNotifications(for: location)
-            }
-        }
+        //This should be done in the model
```

Comment:
Thanks for clarifying why this bit is commented out - Can you update this with a TODO for the next phase of changes?